test/operator-convert.spec.ts

```diff
@@ -21,40 +22,33 @@ const item = {
 
 describe('convert operator unit tests', () => {
   it('it should validate options', () => {
-    expect(() => new ConvertOperator({
+    expectValid({
       name: 'convert', force: true, properties: 'quantity', type: 'int',
-    }).validate())
-      .to.not.throw();
-    expect(() => new ConvertOperator({
+    });
+    expectValid({
       name: 'convert', properties: 'salePrice', preserveArray: true, type: 'int',
-    }).validate())
-      .to.not.throw();
-    expect(() => new ConvertOperator({
+    });
+    expectValid({
       name: 'convert', properties: 'quantity', type: 'int', index: 1,
-    }).validate())
-      .to.not.throw();
-    expect(() => new ConvertOperator({ name: 'convert', properties: 'price,tax', type: 'real' }).validate())
-      .to.not.throw();
-    expect(() => new ConvertOperator({ name: 'convert', properties: ['price', 'tax'], type: 'real' }).validate())
-      .to.not.throw();
-    // @ts-ignore
-    expect(() => new ConvertOperator({ name: 'convert', properties: ['price', 'tax'], type: 'array' }).validate())
-      .to.throw();
-    // @ts-ignore
-    expect(() => new ConvertOperator({ name: 'convert', type: 'array' }).validate())
-      .to.throw();
-    // @ts-ignore
-    expect(() => new ConvertOperator({ name: 'convert', properties: 'quantity' }).validate())
-      .to.throw();
-    expect(() => new ConvertOperator({
-      // @ts-ignore
+    });
+    expectValid({ name: 'convert', enumerate: true });
+    expectValid({
+      name: 'convert', enumerate: true, properties: 'available', type: 'bool',
+    });
+    expectValid({ name: 'convert', properties: 'price,tax', type: 'real' });
+    expectValid({ name: 'convert', properties: ['price', 'tax'], type: 'real' });
+
+    expectInvalid({ name: 'convert' }, 'properties or enumerate or both is needed');
+    expectInvalid({ name: 'convert', properties: 'price' }, 'must also specify type');
+    expectInvalid({ name: 'convert', type: 'real' }, 'must specify properties');
+    expectInvalid({ name: 'convert', enumerate: 'true' }, 'enumerate should be a bool');
+    expectInvalid({ name: 'convert', properties: ['price', 'tax'], type: 'array' }, 'array is not valid');
+    expectInvalid({
```

Comment:
Are there conflicting combinations of properties that we should error or warn about?

Reply:
I _think_ I have them all accounted for.  The one gray area is probably `force` with `enumerate`.  I'll add a card to track.